### ISSUE TITLE
box86 for AArch64 without AArch32 support?

### ISSUE THREAD

**Issue by defrag257** [ptitSeb/box86#528]
Apple Sillicon M1 and some ARMv9-A cores (including Cortex-X2 and Cortex-A510, not including Cortex-A710) has removed AArch32 support. So would you write a box86 for these platforms?

**Comment by ptitSeb** [comment]
box86 will not work on Arrch64 only platform. It's a design restriction. It could, maybe, work with a Aarch32 emulator for Aarch64, but there is no opensource solution for this, to my knowledge.

I have in mind "box32" to address the "run x86 apps on 64bits platforms", but I haven't started it yet, it's still in design phase, and might or might not be written, as I'm still evaluating the technical solutions to create that product.

**Comment by asdfugil** [comment]
Couldn't qemu user mode runs aarch32 on aarch64?

**Comment by Trass3r** [comment]
FEX-Emu should do that.

**Comment by ptitSeb** [comment]
Yes it does: https://box86.org/2022/03/box86-box64-vs-qemu-vs-fex-vs-rosetta2/

**Comment by ptitSeb** [comment]
I have started "box32", but it's a slow dev., so there will be no public release for a long time

**Comment by DarkShadow44** [comment]
> I have started "box32", but it's a slow dev., so there will be no public release for a long time

Does this also mean that you won't make the code public until it's done? Because I'd surely be curious to look at what you came up with already :relaxed:

**Comment by ptitSeb** [comment]
> > I have started "box32", but it's a slow dev., so there will be no public release for a long time
> 
> Does this also mean that you won't make the code public until it's done? Because I'd surely be curious to look at what you came up with already ☺️

Yep. I prefer to not go public until it's actualy usable.

**Comment by etienne0x7C1** [comment]
does it mean rosetta 2 is doing that "box32" work? (as it is able to run steam games)

**Comment by ptitSeb** [comment]
Yes and no.

Rosetta2 support 32bits code, but 32bits libs support has been removed from macOS a log time ago. No 32bits program can be run from steam macOS on the M1.

**Comment by etienne0x7C1** [comment]
hmm I see. Wonder why steam is not providing a 64 bits version only (then we could run it using box64 and also avoid enabling multiarch packages on linux to run it).
And for supporting older games using 32bits, provide something like an 32bit addon..
